File: backend/src/main/java/de/nordbyte/mavazihub/order/entity/Order.java
package de.nordbyte.mavazihub.order.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "orders")
@Getter
@Setter
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(name = "customer_id", nullable = false)
    private UUID customerId;

    @Column(nullable = false, length = 50)
    private String status;

    @Column(name = "payment_status", nullable = false, length = 50)
    private String paymentStatus;

    // Lieferadresse als Snapshot (ADR-05)
    @Column(nullable = false, length = 255)
    private String street;

    @Column(name = "zip_code", nullable = false, length = 20)
    private String zipCode;

    @Column(nullable = false, length = 100)
    private String city;

    @Column(name = "total_price", nullable = false, precision = 10, scale = 2)
    private BigDecimal totalPrice;

    @Column(name = "order_date", nullable = false, updatable = false)
    private LocalDateTime orderDate;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> items = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        this.orderDate = LocalDateTime.now();
    }
}

File: frontend/src/types/Order.ts
export type OrderSummary = {
  id: string
  orderNumber?: string | null
  status?: string | null
  paymentStatus?: string | null
  orderDate?: string | null
  createdAt?: string | null
  orderedAt?: string | null
  totalAmount?: number | null
  totalPrice?: number | null
  itemCount?: number | null
}

export type OrderDetail = OrderSummary & {
  items?: OrderItem[]
  customerId?: string
  street?: string | null
  zipCode?: string | null
  city?: string | null
}

export type OrderItem = {
  id: string
  productId?: number | null
  productName?: string | null
  quantity: number
  unitPrice?: number | null
  totalPrice?: number | null
  returnableQuantity?: number | null
}

export type CheckoutRequest = {
  street: string
  zipCode: string
  city: string
}
